qiskit/compiler/transpile.py

```diff
@@ -234,42 +206,36 @@ def _parse_transpile_args(circuits, backend,
     arg has more priority than the arg set by backend)
 
     Returns:
-        list[TranspileConfig]: a transpile config for each circuit, which is a standardized
-            object that configures the transpiler and determines the pass manager to use.
+        list[dicts]: a list of transpile parameters.
     """
     # Each arg could be single or a list. If list, it must be the same size as
     # number of circuits. If single, duplicate to create a list of that size.
     num_circuits = len(circuits)
 
     basis_gates = _parse_basis_gates(basis_gates, backend, circuits)
-
-    coupling_map = _parse_coupling_map(coupling_map, backend, num_circuits)
-
+    coupling_map = _parse_coupling_map(coupling_map, backend, circuits)
     backend_properties = _parse_backend_properties(backend_properties, backend, num_circuits)
-
     initial_layout = _parse_initial_layout(initial_layout, circuits)
-
     seed_transpiler = _parse_seed_transpiler(seed_transpiler, num_circuits)
-
     optimization_level = _parse_optimization_level(optimization_level, num_circuits)
-
     pass_manager = _parse_pass_manager(pass_manager, num_circuits)
-
     output_name = _parse_output_name(output_name, circuits)
 
     transpile_configs = []
     for args in zip(basis_gates, coupling_map, backend_properties,
                     initial_layout, seed_transpiler, optimization_level,
-                    pass_manager, output_name):
-        transpile_config = TranspileConfig(basis_gates=args[0],
-                                           coupling_map=args[1],
-                                           backend_properties=args[2],
-                                           initial_layout=args[3],
-                                           seed_transpiler=args[4],
-                                           optimization_level=args[5],
-                                           pass_manager=args[6],
-                                           callback=callback,
-                                           output_name=args[7])
+                    pass_manager, output_name, circuits):
+        transpile_config = {'pass_manager_config': PassManagerConfig(basis_gates=args[0],
+                                                                     coupling_map=args[1],
+                                                                     backend_properties=args[2],
+                                                                     initial_layout=args[3],
+                                                                     seed_transpiler=args[4],
+                                                                     callback=callback),
+                            'optimization_level': args[5],
+                            'pass_manager': args[6],
+                            'output_name': args[7],
+                            'circuit': args[8],
+                            'callback': callback}
```

Comment:
Is `callback` needed in both `transpile_args` and `PassManagerConfig`?

Reply:
removed in fe54db62572bd095ef85c52e47d48794c8cd235d